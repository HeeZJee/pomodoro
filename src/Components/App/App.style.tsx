import styled from "styled-components";

export const Wrapper = styled.div`
  width: 100vw;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  background: rgb(0, 255, 158);
  background: linear-gradient(
    45deg,
    rgb(0, 191, 111) 0%,
    rgb(0, 145, 255) 50%,
    rgb(110, 0, 189) 100%
  );
`;
