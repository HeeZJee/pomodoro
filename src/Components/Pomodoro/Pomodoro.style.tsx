import styled from "styled-components";

export const Wrapper = styled.div`
  width: 50vw;
  height: 70vh;
  display: flex;
  background-color: #efefef;
  border-radius: 10px;
  box-shadow: 0 0 50px 0 rgba(0, 0, 0, 0.5);

  @media screen and (max-width: 960px) {
    width: 70vh;
  }

  @media screen and (max-width: 510px) {
    width: 60vh;
  }
  @media screen and (max-width: 400px) {
    width: 50vh;
  }
`;
