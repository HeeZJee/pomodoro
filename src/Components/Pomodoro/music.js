const songsCollections = [
  "http://www.music.helsinki.fi/tmt/opetus/uusmedia/esim/a2002011001-e02.wav",
  "http://www.noiseaddicts.com/samples_1w72b820/2563.mp3",
  "http://www.noiseaddicts.com/samples_1w72b820/2558.mp3",
];

const getRandomFromRange = (min, max) => {
  return Math.floor(Math.random() * (max - min) + min);
};

const randomSongIndex = getRandomFromRange(0, songsCollections.length);

console.log("Random song index is ", randomSongIndex);

const randomSong = songsCollections[randomSongIndex];
new Audio(randomSong).play();
