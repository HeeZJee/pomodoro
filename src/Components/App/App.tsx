import React from "react";
import Pomodoro from "../Pomodoro/Pomodoro";
import { Wrapper } from "./App.style";

function App() {
  return (
    <Wrapper data-testid="App">
      <Pomodoro />
    </Wrapper>
  );
}

export default App;
