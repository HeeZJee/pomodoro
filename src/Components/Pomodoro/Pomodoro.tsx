import React, { useState } from "react";
import Button from "../../Button/Button";
import Timer from "../TImer/Timer";
import { ButtonsWrapper, Wrapper } from "./Pomodoro.style";
import { FaPlay, FaStop, FaRedoAlt, FaMusic } from "react-icons/fa";

interface iState {
  minute: number;
  second: number;
  isOn: boolean;
  playMusic: boolean;
}

const initialState = {
  minute: 25,
  second: 0,
  isOn: true,
  playMusic: false,
};

const Pomodoro = () => {
  const [state, setState] = useState<iState>(initialState);
  const { minute, second, isOn, playMusic } = state;

  return (
    <Wrapper data-testid="pomodoro">
      <Timer minute={minute} second={second} />
      <ButtonsWrapper data-testid="buttons">
        <Button
          title="Play music"
          icon={<FaMusic color="rgb(106, 89, 157)" size={18} />}
          active={playMusic}
        />
        {!isOn ? (
          <Button title="Start pomodoro" text="Start" icon={<FaPlay />} />
        ) : (
          <Button title="Stop pomodoro" text="Stop" icon={<FaStop />} />
        )}
        <Button
          title="Reset pomodoro"
          icon={<FaRedoAlt color="rgb(106, 89, 157)" size={18} />}
        />
      </ButtonsWrapper>
    </Wrapper>
  );
};

export default Pomodoro;
