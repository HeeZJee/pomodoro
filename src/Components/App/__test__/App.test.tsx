import React from "react";
import { render } from "@testing-library/react";
import App from "./../App";
import Pomodoro from "../../Pomodoro/Pomodoro";

it("renders App", () => {
  const { container } = render(<App />);
  expect(container.childElementCount).toBeGreaterThanOrEqual(1);
});

it("snapshot App", () => {
  const { getByTestId } = render(<App />);
  expect(getByTestId("App")).toMatchSnapshot();
});
