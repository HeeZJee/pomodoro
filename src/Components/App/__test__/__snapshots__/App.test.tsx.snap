// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot App 1`] = `
<div
  class="sc-gKsewC kgAwfb"
  data-testid="App"
>
  <div
    class="sc-eCssSg gMrqqQ"
    data-testid="pomodoro"
  >
    <div
      class="sc-hKgILt bHHqfk"
      data-testid="timer"
    >
      <div
        data-test-id="CircularProgressbarWithChildren"
      >
        <div
          style="position: relative; width: 100%; height: 100%;"
        >
          <svg
            class="CircularProgressbar "
            data-test-id="CircularProgressbar"
            viewBox="0 0 100 100"
          >
            <path
              class="CircularProgressbar-trail"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fill-opacity="0"
              stroke-width="8"
              style="stroke: rgba(106, 89, 157, 0.3); stroke-linecap: round; stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 0px;"
            />
            <path
              class="CircularProgressbar-path"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fill-opacity="0"
              stroke-width="8"
              style="stroke: rgba(106, 89, 157, 1); stroke-linecap: round; transition: none; transition-duration: 60s; stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 11.561060965210448px;"
            />
          </svg>
          <div
            data-test-id="CircularProgressbarWithChildren__children"
            style="position: absolute; width: 100%; height: 100%; margin-top: -100%; display: flex; flex-direction: column; justify-content: center; align-items: center;"
          >
            <p>
              <span
                data-testid="minute"
              >
                25
              </span>
              :
              <span
                data-testid="second"
              >
                0
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-jSgupP flLuNI"
    >
      <button
        class="sc-dlfnbm gcAlKd"
        title="Play music"
      >
        <svg
          color="rgb(106, 89, 157)"
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(106, 89, 157);"
          viewBox="0 0 512 512"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M470.38 1.51L150.41 96A32 32 0 0 0 128 126.51v261.41A139 139 0 0 0 96 384c-53 0-96 28.66-96 64s43 64 96 64 96-28.66 96-64V214.32l256-75v184.61a138.4 138.4 0 0 0-32-3.93c-53 0-96 28.66-96 64s43 64 96 64 96-28.65 96-64V32a32 32 0 0 0-41.62-30.49z"
          />
        </svg>
      </button>
      <button
        class="sc-bdfBwQ lcmLlu"
        title="Stop pomodoro"
      >
        <span
          class="sc-gsTCUz edqCiE"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
            />
          </svg>
        </span>
        <span>
          Stop
        </span>
      </button>
      <button
        class="sc-dlfnbm gcAlKd"
        title="Reset pomodoro"
      >
        <svg
          color="rgb(106, 89, 157)"
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(106, 89, 157);"
          viewBox="0 0 512 512"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256.455 8c66.269.119 126.437 26.233 170.859 68.685l35.715-35.715C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.75c-30.864-28.899-70.801-44.907-113.23-45.273-92.398-.798-170.283 73.977-169.484 169.442C88.764 348.009 162.184 424 256 424c41.127 0 79.997-14.678 110.629-41.556 4.743-4.161 11.906-3.908 16.368.553l39.662 39.662c4.872 4.872 4.631 12.815-.482 17.433C378.202 479.813 319.926 504 256 504 119.034 504 8.001 392.967 8 256.002 7.999 119.193 119.646 7.755 256.455 8z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
